cfme/common/provider.py

```diff
@@ -404,12 +426,20 @@ def _do_stats_match(self, client, stats_to_match=None, refresh_timer=None, ui=Fa
 
     @property
     def exists(self):
-        ems = cfmedb()['ext_management_systems']
-        provs = (prov[0] for prov in cfmedb().session.query(ems.name))
-        if self.name in provs:
+        """ Checks if exists among managed providers
+
+        Note:
+            If provider exists but with a different name, name is updated
+        """
+        for prov in store.current_appliance.managed_providers:
```

Comment:
Still not sure I understand the problem. I assume we just pass the context through to the appliance. Is that not possible?